### PR TITLE
propagate getNodeFromDB err

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**

### CODE REVIEW

File: builtInFunctions/multiESDTNFTTransfer.go [hunk 1]
```diff
@@ -297,6 +297,10 @@ func (e *esdtNFTMultiTransfer) processESDTNFTMultiTransferOnSenderShard(
 			dstAddress,
 			listTransferData[i],
 			vmInput.ReturnCallAfterError)
+		if core.IsGetNodeFromDBError(err) {
+			log.Error("processESDTNFTMultiTransferOnSenderShard", "error", err, "token", string(listTransferData[i].ESDTTokenName))
```
Comment (iulianpascalau) [inline builtInFunctions/multiESDTNFTTransfer.go:301]: do we need this log here? We either return error or log it. Maybe a wrapping will help?

File: go.mod [hunk 2]
```diff
@@ -4,7 +4,7 @@ go 1.17
 
 require (
 	github.com/mitchellh/mapstructure v1.4.1
-	github.com/multiversx/mx-chain-core-go v1.2.0
+	github.com/multiversx/mx-chain-core-go v1.2.1-0.20230403113932-916b16d18978
```
Comment (iulianpascalau) [inline go.mod:7]: proper release

File: builtInFunctions/esdtDataStorage.go [hunk 3]
```diff
@@ -131,6 +131,9 @@ func (e *esdtDataStorage) getESDTNFTTokenOnDestinationWithAccountsAdapterOptions
 		Type:  uint32(core.Fungible),
 	}
 	marshaledData, _, err := accnt.AccountDataHandler().RetrieveValue(esdtNFTTokenKey)
+	if core.IsGetNodeFromDBError(err) {
```
Comment (iulianpascalau) [inline builtInFunctions/esdtDataStorage.go:134]: let's add unit tests for all the new if branches

**Review by sasurobert — APPROVED** [review]

File: builtInFunctions/multiESDTNFTTransfer.go (hunk 1 above)
Comment (BeniaminDrasovean) [reply builtInFunctions/multiESDTNFTTransfer.go:301]: Removed the log print. No wrapping needed.

File: builtInFunctions/esdtDataStorage.go (hunk 3 above)
Comment (BeniaminDrasovean) [reply builtInFunctions/esdtDataStorage.go:134]: Added.

File: go.mod (hunk 2 above)
Comment (BeniaminDrasovean) [reply go.mod]: Will do before merge.

File: go.mod [hunk 4]
```diff
@@ -4,7 +4,7 @@ go 1.17
 
 require (
 	github.com/mitchellh/mapstructure v1.4.1
-	github.com/multiversx/mx-chain-core-go v1.2.1-0.20230403113932-916b16d18978
+	github.com/multiversx/mx-chain-core-go v1.2.3-0.20230512130104-2a2e00c016b2
```
Comment (iulianpascalau) [inline go.mod:7]: just the proper release

**Review by sasurobert — DISMISSED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**